examples/algorithms/bc/bc.cu

```diff
@@ -1,32 +1,100 @@
 #include <gunrock/algorithms/bc.hxx>
+#include "gunrock/util/performance.hxx"
+#include <cxxopts.hpp>
 
 using namespace gunrock;
 using namespace memory;
 
-void test_bc(int num_arguments, char** argument_array) {
-  if (num_arguments != 2) {
-    std::cerr << "usage: ./bin/<program-name> filename.mtx" << std::endl;
-    exit(1);
+struct parameters_t {
+  std::string filename;
+  std::string json_dir = ".";
+  std::string json_file = "";
+  int num_runs = 1;
+  cxxopts::Options options;
+  bool performance = false;
+  bool binary = false;
+
+  /**
+   * @brief Construct a new parameters object and parse command line arguments.
+   *
+   * @param argc Number of command line arguments.
+   * @param argv Command line arguments.
+   */
+  parameters_t(int argc, char** argv)
+      : options(argv[0], "Breadth First Search example") {
+    // Add command line options
+    options.add_options()("help", "Print help")  // help
+        ("performance", "performance analysis")  // performance evaluation
+        ("m,market", "Matrix file", cxxopts::value<std::string>())  // mtx file
+        ("n,num_runs", "Number of runs", cxxopts::value<int>())     // runs
+        ("d,json_dir", "JSON output directory",
+         cxxopts::value<std::string>())  // json output directory
+        ("f,json_file", "JSON output file",
+         cxxopts::value<std::string>());  // json output file
+
+    // Parse command line arguments
+    auto result = options.parse(argc, argv);
+
+    if (result.count("help") || (result.count("market") == 0)) {
+      std::cout << options.help({""}) << std::endl;
+      std::exit(0);
+    }
+
+    if (result.count("market") == 1) {
+      filename = result["market"].as<std::string>();
+      if (util::is_binary_csr(filename)) {
+        binary = true;
+      } else if (!util::is_market(filename)) {
+        std::cout << options.help({""}) << std::endl;
+        std::exit(0);
+      }
+    } else {
+      std::cout << options.help({""}) << std::endl;
+      std::exit(0);
+    }
+
+    if (result.count("performance") == 1) {
+      performance = true;
+    }
+
+    if (result.count("num_runs") == 1) {
+      num_runs = result["num_runs"].as<int>();
+    }
+
+    if (result.count("json_dir") == 1) {
+      json_dir = result["json_dir"].as<std::string>();
+    }
+
+    if (result.count("json_file") == 1) {
+      json_dir = result["json_file"].as<std::string>();
+    }
   }
+};
```

Comment:
I noticed that this bit of code is _almost_ exactly the same in bfs, pr and sssp .cu files. Is there a way you could allow `parameters_t` constructor to take the name of the algorithm, unify all of these under `include/gunrock/io/parameters.hxx`, and simply include that header file in all 4 examples? This should reduce the code significantly and promote reuse.

Also, feel free to assign it a namespace, like:
```cpp
namespace gunrock { // main namespace
namespace io {  // input/output main namespace
namespace cli { // maybe?
```